drivers/simulator.py

```diff
@@ -0,0 +1,234 @@
+                            "10001",
+                            "11111"]
+        # Data for custom character #6. Code {0x05}
+        self.char_6_data = ["11111",
+                            "10001",
+                            "10001",
+                            "10001",
+                            "10001",
+                            "10001",
+                            "10001",
+                            "11111"]
+        # Data for custom character #7. Code {0x06}
+        self.char_7_data = ["11111",
+                            "10001",
+                            "10001",
+                            "10001",
+                            "10001",
+                            "10001",
+                            "10001",
+                            "11111"]
+        # Data for custom character #8. Code {0x07}
+        self.char_8_data = ["11111",
+                            "10001",
+                            "10001",
+                            "10001",
+                            "10001",
+                            "10001",
+                            "10001",
+                            "11111"]
+
+    # load custom character data to CG RAM for later use in extended string. Data for
+    # characters is hold in file custom_characters.txt in the same folder as i2c_dev.py
+    # file. These custom characters can be used in printing of extended string with a
+    # placeholder with desired character codes: 1st - {0x00}, 2nd - {0x01}, 3rd - {0x02},
+    # 4th - {0x03}, 5th - {0x04}, 6th - {0x05}, 7th - {0x06} and 8th - {0x07}.
+    def load_custom_characters_data(self):
+        char_data_list = [
+            (f"{{0x00}}", self.char_1_data),
+            (f"{{0x01}}", self.char_2_data),
+            (f"{{0x02}}", self.char_3_data),
+            (f"{{0x03}}", self.char_4_data),
+            (f"{{0x04}}", self.char_5_data),
+            (f"{{0x05}}", self.char_6_data),
+            (f"{{0x06}}", self.char_7_data),
+            (f"{{0x07}}", self.char_8_data)
+        ]
+
+        for char_name, bitmap in char_data_list:
+            if len(bitmap) != 8 or any(len(row) != 5 for row in bitmap):
+                continue
+            custom_chars[char_name] = bitmap
+
+    def get_custom_char(self, char_name):
+        return custom_chars.get(char_name, ["00000"] * 8)
+
+    # Draw CustomCharacters
+    def draw_custom_char(self, bitmap, x, y, color):
+        pixel_size = 15
+        for row, line in enumerate(bitmap):
+            for col, bit in enumerate(line):
+                if bit == '1':
+                    rect_id = self.lcd.canvas.create_rectangle(
+                        x + (col * pixel_size),
+                        y + (row * pixel_size),
+                        x + ((col + 1) * pixel_size),
+                        y + ((row + 1) * pixel_size),
+                        fill=color,
+                        outline=color
+                    )
+                    self.lcd.rectangles.append(rect_id)
```

Comment:
i couldn't help but notice that this implementation is re-using logic from the original `CustomCharacters` class. can't you inherit the original and override where necessary?